src/allmydata/node.py

```diff
@@ -170,11 +178,11 @@ class _Config(object):
     as a helper instead.
     """
 
-    def __init__(self, configparser, portnum_fname, config_fname):
+    def __init__(self, configparser, portnum_fname, basedir, config_fname):
```

Comment:
Not sure if Tahoe-LAFS has API documentation requirements but I would like it if it did.  It would be nice to see API docs explaining all these parameters - but at least the new `basedir` parameter.

Through the rest of this PR, similar comments could be made.  I won't make them repeatedly but if you agree that API documentation is good, please update the other areas of code where a new API is being added or a new parameter is being added to an existing API.